# Housing Market

### Introduction:

This time we will create our own dataset with fictional numbers to describe a house market. As we are going to create random data don't try to reason of the numbers.

### Step 1. Import the necessary libraries

In [2]:
import pandas as pd
import numpy as np

### Step 2. Create 3 differents Series, each of length 100, as follows: 
1. The first a random number from 1 to 4 
2. The second a random number from 1 to 3
3. The third a random number from 10,000 to 30,000

In [5]:
# Parameters
# lowint or array-like of ints
# Lowest (signed) integers to be drawn from the distribution (unless high=None, in which case this parameter is one above the highest such integer).

# highint or array-like of ints, optional
# If provided, one above the largest (signed) integer to be drawn from the distribution (see above for behavior if high=None). If array-like, must contain integer values

# sizeint or tuple of ints, optional
# Output shape. If the given shape is, e.g., (m, n, k), then m * n * k samples are drawn. Default is None, in which case a single value is returned.

# dtypedtype, optional
# Desired dtype of the result. Byteorder must be native. The default value is int.

# New in version 1.11.0.

# Returns
# outint or ndarray of ints
# size-shaped array of random integers from the appropriate distribution, or a single such random int if size not provided.

s1 = pd.Series(np.random.randint(1,high=5,size=100))
s2 = pd.Series(np.random.randint(1,high=4,size=100))
s3 = pd.Series(np.random.randint(10000,high=30000,size=100))

### Step 3. Let's create a DataFrame by joinning the Series by column

In [14]:
df = pd.concat([s1,s2,s3], axis=1) #concat along the col?
df.head()

0  1      2
0  3  1  29865
1  3  3  20196
2  3  1  22447
3  4  3  19925
4  4  1  29446

### Step 4. Change the name of the columns to bedrs, bathrs, price_sqr_meter

In [16]:
df = df.rename(columns={0: 'bedrs',1:'bathrs',2:'price_sqr_meter'})
df.head()

bedrs  bathrs  price_sqr_meter
0      3       1            29865
1      3       3            20196
2      3       1            22447
3      4       3            19925
4      4       1            29446

### Step 5. Create a one column DataFrame with the values of the 3 Series and assign it to 'bigcolumn'

In [23]:
bigcolumn = pd.concat([s1,s2,s3],axis =0) # concat along the rows
#converts series to df
bigcolumn= bigcolumn.to_frame()


### Step 6. Oops, it seems it is going only until index 99. Is it true?

In [22]:
bigcolumn.head(-1)

0
0       3
1       3
2       3
3       4
4       4
..    ...
94  14186
95  27538
96  29748
97  24048
98  10177

[299 rows x 1 columns]

### Step 7. Reindex the DataFrame so it goes from 0 to 299

In [26]:
# what does drop do?
bigcolumn.reset_index(drop=True,inplace=True)
bigcolumn.head(-1)

level_0  index      0
0          0      0      3
1          1      1      3
2          2      2      3
3          3      3      4
4          4      4      4
..       ...    ...    ...
294      294     94  14186
295      295     95  27538
296      296     96  29748
297      297     97  24048
298      298     98  10177

[299 rows x 3 columns]